bioimageio/spec/model/v0_4/schema.py

```diff
@@ -456,23 +456,36 @@ def get_min_shape(t) -> numpy.ndarray:
             if isinstance(t.shape, raw_nodes.ParametrizedInputShape):
                 shape = numpy.array(t.shape.min)
             elif isinstance(t.shape, raw_nodes.ImplicitOutputShape):
-                shape = get_min_shape(tensors_by_name[t.shape.reference_tensor]) * t.shape.scale + 2 * numpy.array(
-                    t.shape.offset
-                )
+                scale = list(t.shape.scale)
+                ref_shape = get_min_shape(tensors_by_name[t.shape.reference_tensor])
+
+                if any(sc is None for sc in scale):
+                    expanded_dims = tuple(idx for idx, sc in enumerate(scale) if sc is None)
+                    new_ref_shape = []
+                    for idx in range(len(scale)):
+                        ref_idx = idx - sum(int(exp < idx) for exp in expanded_dims)
+                        new_ref_shape.append(1 if idx in expanded_dims else ref_shape[ref_idx])
+                    ref_shape = numpy.array(new_ref_shape)
+                    assert len(ref_shape) == len(scale)
+                    scale = [0.0 if sc is None else sc for sc in scale]
+
+                offset = numpy.array(t.shape.offset)
+                shape = ref_shape * numpy.array(scale) + 2 * offset
             else:
                 shape = numpy.array(t.shape)
 
             return shape
 
         for out in outs:
-            if isinstance(out.shape, raw_nodes.ImplicitOutputShape) and len(out.shape) != len(
-                tensors_by_name[out.shape.reference_tensor].shape
-            ):
-                raise ValidationError(
-                    f"Referenced tensor {out.shape.reference_tensor} "
-                    f"with {len(tensors_by_name[out.shape.reference_tensor].shape)} dimensions does not match "
-                    f"output tensor {out.name} with {len(out.shape)} dimensions."
-                )
+            if isinstance(out.shape, raw_nodes.ImplicitOutputShape):
+                ndim_ref = len(tensors_by_name[out.shape.reference_tensor].shape)
+                ndim_out = len([scale for scale in out.shape.scale if scale is not None])
```

Comment:
here `ndim_out` would include the `None` axis, but you compare to `ndim_ref`, so you leave it out.. maybe we can rename this here or comment on it?